### PR TITLE
Added switches to the admin module User Status (mod_status)

### PR DESCRIPTION
**Pull request by RickR2H**
Changes:
1. Added switches and make the link parameters work in the Administrator module "mod_status" so the links can be switched on and off.
2. Updated the language file.
3. The second switch is added as extra link to open a new Administrator window for easy Joomla management.

Test instruction:
1. Go to the Module manager in the Administrator and set the top filter to Administrator
2. Open the module "User Status"
3. Turn on and off the "Show view site" and "Show view Administrator" switches and check if both links disappear and reappear in the bottom of the site (See screenshot).

Click the "Show Site" and the "Show Admin" links in the bottom of the Administrator area and check if they open a new Site window and an new Administrator Window. The links should also work when editing and article (See screenshot).

footer-links-site-admin-joomla

### CODE REVIEW

**Comment by roland-d** [comment]
@RickR2H The issue is not in the PR but in Travis itself, I will see what I can do about it.

**Comment by astridx** [comment]
1.	I applied the patch and I saw the "Show Site" and the "Show Admin" links in the bottom of the Administrator area (snap1).
2.	I went to the Module manager in the Administrator and set the top filter to Administrator
3.	I opened the module "User Status" and saw the two new Buttons. They were activated.
4.	I turned off and on the "Show view site" and "Show view Administrator" switches and check if both links disappear and reappear in the bottom of the site after saving. (snap2 and snap3).
5.	I clicked the "Show Site" and the "Show Admin" links in the bottom of the Administrator area while they were activated. They open a new Site window/ a new Administrator Window. 
6.	The links worked while editing and article, too.
screen shot 2015-06-15 at 02 07 09screen shot 2015-06-15 at 02 07 12screen shot 2015-06-15 at 02 07 12<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7174">issues.joomla.org/joomla-cms/7174</a>.</sub>

**Comment by RichardR2H** [comment]
@test Succesfully tested and works!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7174">issues.joomla.org/joomla-cms/7174</a>.</sub>

**Comment by ghazal** [comment]
@test 
Nice new feature. Tks.
Works OK here.

File: administrator/language/en-GB/en-GB.mod_status.ini [hunk 1]
```diff
@@ -7,6 +7,11 @@ MOD_STATUS="User Status"
 MOD_STATUS_BACKEND_USERS_0="Admins"
 MOD_STATUS_BACKEND_USERS_1="Admin"
 MOD_STATUS_BACKEND_USERS_MORE="Admins"
+MOD_STATUS_FIELD_SHOW_VIEWSITE_LABEL="Show View Site"
+MOD_STATUS_FIELD_SHOW_VIEWSITE_DESC="Show a link to the website homepage."
+MOD_STATUS_FIELD_LINK_VIEWSITE_LABEL="Show Admin"
```
Comment (Bakual) [inline administrator/language/en-GB/en-GB.mod_status.ini]: Shouldn't that key be named something like `MOD_STATUS_FIELD_LINK_VIEWADMIN_LABEL` as it opens a new admin window, not site?

**Comment by Bakual** [comment]
I can see a small use of a parameter for the "View Site" link. Especially since the code already checks for that (not existing) parameter.

I don't really see the use for a link to open a new admin window. The exact same can be achieved by simply ctrl-clicking the Joomla icon in the top left corner. At least on Windows. On other OS it's likely similar simple.

**Comment by RickR2H** [comment]
@Bakual Regarding the language string you've got a point! I'll change it accordingly.

**Comment by RickR2H** [comment]
@Bakual The top left click doesn't work when an article is opened!

As a regular web editor I'm coping a lot of article text/code back and forth. When I've opened an article, I have to copy the admin URL and paste it in a new window to open an other article or module. I thought this extra link would make this process a little easier.

**Comment by MSnoeren1995** [comment]
@test Works! Nice new feature!

**Comment by Bakual** [comment]
> The top left click doesn't work when an article is opened!

I'll give you that. Thus usecase is found. :smile:

**Comment by Bakual** [comment]
Setting RTC based on various tests.
Since we are currently in language freeze for 3.4.2, this has to wait or go into 3.5.0.

**Comment by RickR2H** [comment]
Nice :smile: My first real contribution will be a fact!

**Comment by Bakual** [comment]
Merged into `3.5-dev` with https://github.com/joomla/joomla-cms/commit/53e6aa9e76cb0d4ed9fadacc6a86e5980c179754.

Thanks @RickR2H and grats for your first PR :+1: 
It will be part of Joomla 3.5 and your named will be mentioned in the release announcement :smile:

**Comment by losedk** [comment]
Sorry for commenting on a merged PR. But why add a link to the admin panel? mod_status is an admin module.

**Comment by Bakual** [comment]
@losedk Rick explained a few posts above.

**Comment by losedk** [comment]
@Bakual Ahh I see, valid argument. But it seems a bit excessive to add yet another link in the footer.

**Comment by brianteeman** [comment]
Sorry but this is the wrong place for this change AND/OR the module needs
to be renamed and be given a new description

User Status
Administrator

This module shows the status of the logged-in users.

On 15 June 2015 at 14:06, Peter Lose <notifications@github.com> wrote:

> @Bakual <https://github.com/Bakual> Ahh I see, valid argument. But it
> seems a bit excessive to add yet another link in the footer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/7174#issuecomment-112063618>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by Bakual** [comment]
@losedk If it's an issue, we could disable it by default. So nothing changes for anyone. But I guess most probably don't notice at all.

**Comment by Bakual** [comment]
> This module shows the status of the logged-in users.

It also showed the link to frontend before this change. So it was already not only about users.
Maybe it's a bit misnamed indeed. The module itself uses `mod_status` as its code-name.

**Comment by brianteeman** [comment]
I know its only a small change but we've been trying to remove stuff from
the core and to make the UI cleaner and simpler.

If we accept every PR because someone has a usecase for it then we will
have the JED inside the core.

In this case it can be achieved either by installing a module or by a
template override.

On 15 June 2015 at 14:19, Thomas Hunziker <notifications@github.com> wrote:

> This module shows the status of the logged-in users.
>
> It also showed the link to frontend before this change. So it was already
> not only about users.
> Maybe it's a bit misnamed indeed. The module itself uses mod_status as
> its code-name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/7174#issuecomment-112066403>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by RickR2H** [comment]
I agree that every change has to be carefully considered. The reason why I made the modifications was that it could open a new site window but not a new admin window which seems strange to me. Don't forget the logout link which is also present on the bottom. 
In the menu bar the three options, "open a new admin window", "open a new site window" and "logout" are all possible in the standard non-editor interface. The whole top menu is disabled when editing a article/module so we can't access these items. In my humble opinion adding the link to the module and make these three items work in the footer seems a logical step. In the process also added the missing option switches to the module.

Switching all the links off in the footer by default could be a good option for the people who need a cleaner interface.

**Comment by brianteeman** [comment]
On 15 Jun 2015 3:18 pm, "Rick Spaan" <notifications@github.com> wrote:
>

>
> Switching all the links off in the footer by default could be a good
option for the people who need a cleaner interface.
>
As is creating a new module available on the JED for those that want the
extra links

**Comment by RickR2H** [comment]
Net time a I carefully consider creating a PR...

**Comment by mbabker** [comment]
Nobody's trying to suggest that you shouldn't contribute.  The reality is for a long time we've accepted a lot of pulls simply because someone took the time to code it and we didn't want to say no.  We really need to get to a point where things are being looked at beyond "does it work", which is probably 95% of what influences whether something is determined to be "ready to commit".

**Comment by losedk** [comment]
Sorry for starting this debate @RickR2H. My intent was not to bash your PR and contribution, it is very appreciated :) I'm merely focused on keeping the UI as clean as possible and I was just a bit puzzled by this being merged so quickly.

**Comment by RickR2H** [comment]
@losedk I don't take it personally and I think it is good to have a debate and be critical about whether a change needs to be in the core or not. As a newbie contributor like me it is difficult to decide whether a change is needed or not as there are no direct guidelines. Also with a lot of (cosmetic)changes usability plays an important role but also personal taste. At the end I hope these discussions will make Joomla a better product...!

**Comment by losedk** [comment]
@RickR2H Great! Hope you will continue to contribute to Joomla and make it even better :)

**Pull request closed without merging**